yellow-paper/docs/gas-and-fees/gas-metering-and-fees.md

```diff
@@ -0,0 +1,683 @@
+---
+title: Gas Metering and Fees
+---
+
+# Gas Metering and Fees
+
+# Disclaimers, assumptions, key points
+
+:::warning
+This document contains proposed changes to the protocol.
+
+Please read it closely.
+:::
+
+### Assumptions
+There are a handful of major assumptions that this document makes, in line with the current state of the protocol:
+
+- There is no enshrined enforcement of fee payment
+- We must support pure private transactions
+- We publish full transaction effects for each transaction
+
+:::warning
+Are these assumptions still valid?
```

Comment:
Not sure if the final one "We publish full transaction effects for each transaction" is completely decided. I think there's still uncertainty around whether Aztec will publish the intents (or inputs) of transactions to L1/DA, or whether Aztec will publish state diffs (the outputs) of transactions to L1/DA.
I think it depends on the sequencer selection and preconfirmation designs, and on analysis of the amount of data that would need to be broadcast in each case, and on the auditability of history in a state-diff version of a chain.

Reply:
Yes!